feature: read MCU data into modals

StefanCorneteanu opened the pull request:
## Description

Added MCU modals to UDS page, similar to the ones on Android. For reading data, I added the read_info_mcu endpoint

## Trello link [here](https://trello.com/c/HRFFr9LP/139-update-mcu-page)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Test (changes related to test code)
- [ ] Refactor (non-breaking change which does not add functionality)
- [ ] Chore (changes unrelated to production/testing code)

## How was this tested? Please provide answer below, or N/A if not applicable (chores only)
As preconditions, start the MCU and REST API. Go to http://localhost:3000/UDS/mcu. The following data should appear:

![Screenshot from 2025-03-21 07-15-36](https://github.com/user-attachments/assets/e43ec58d-45d3-4d90-be8e-b54750402c5e)

AlexZahariaRandstad commented on carsdata/src/utils/HexConverter.ts:
```diff
@@ -0,0 +1,27 @@
+    else{
+        return "Invalid String"
+    }
+}
```
Add a newline at the end of file to avoid GitHub warning.

StefanCorneteanu replied:
Done

AlexZahariaRandstad reviewed (changes requested):
Currently, the left-side boxes are not well aligned. Maybe a quick fix to align them properly?

StefanCorneteanu commented:
> Currently, the left-side boxes are not well aligned. Maybe a quick fix to align them properly?

Reworked alignment. Updated the screenshot in the PR
